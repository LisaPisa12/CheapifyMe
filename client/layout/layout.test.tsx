import AppLayout from './layout';

import { cleanup, render, screen } from '@testing-library/react';
import { ReactNode } from 'react';

const renderComponent = (children: ReactNode) =>
  render(<AppLayout>{children}</AppLayout>);

describe('App layout', () => {
  afterEach(() => {
    cleanup();
  });

  it('Renders the layout', () => {
    const component = renderComponent(<p>Hello</p>);
    expect(component.baseElement).toMatchSnapshot();
  });

  it('Renders the center div', () => {
    renderComponent(<p>Hello</p>);
    expect(screen.queryByTestId('center-div')).toBeTruthy();
  });

  it('Renders the main', () => {
    renderComponent(<p>Hello</p>);
    expect(screen.queryByTestId('main')).toBeTruthy();
  });

  it('Renders children elements', async () => {
    renderComponent(<p>LisaPisa</p>);
    expect(screen.queryByText(/LisaPisa/i)).toBeTruthy();
  });
});
