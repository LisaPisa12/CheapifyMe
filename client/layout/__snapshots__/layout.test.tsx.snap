// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App layout Renders the layout 1`] = `
<body>
  <div>
    <div
      class="center"
      data-testid="center-div"
    >
      <main
        class="main"
        data-testid="main"
      >
        <p>
          Hello
        </p>
      </main>
    </div>
  </div>
</body>
`;
